lib/oban/job.ex

```diff
@@ -366,12 +366,34 @@ defmodule Oban.Job do
     end
   end
 
+  @doc false
+  @spec cast_period(pos_integer() | {atom(), pos_integer()}) :: pos_integer()
+  def cast_period({value, unit}) do
+    unit = to_string(unit)
+
+    cond do
+      unit in ~w(second seconds) -> value
+      unit in ~w(minute minutes) -> value * 60
+      unit in ~w(hour hours) -> value * 60 * 60
+      unit in ~w(day days) -> value * 24 * 60 * 60
+      true -> unit
+    end
+  end
+
+  def cast_period(period), do: period
+
+  @valid_period_units ~w(second seconds minute minutes hour hours day days)a
+
   @doc false
   @spec valid_unique_opt?({:fields | :period | :states, [atom()] | integer()}) :: boolean()
   def valid_unique_opt?({:fields, [_ | _] = fields}), do: fields -- [:meta | @unique_fields] == []
   def valid_unique_opt?({:keys, []}), do: true
   def valid_unique_opt?({:keys, [_ | _] = keys}), do: Enum.all?(keys, &is_atom/1)
   def valid_unique_opt?({:period, :infinity}), do: true
+
+  def valid_unique_opt?({:period, {period, unit}}),
+    do: is_integer(period) and period > 0 and unit in @valid_period_units
```

Comment:
Nitpick, but we don't use this style anywhere else:

```suggestion
  def valid_unique_opt?({:period, {period, unit}}) do
    is_integer(period) and period > 0 and unit in @valid_period_units
  end
```